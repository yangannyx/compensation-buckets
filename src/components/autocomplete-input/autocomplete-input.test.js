import { shallow } from 'enzyme';
import { AutocompleteInput, DropdownItem } from './';

describe('AutocompleteInput component', () => {
  it('should render', () => {
    // TODO: add render snapshot test
    const wrapper = shallow(<AutocompleteInput />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('DropdownItem component', () => {
  it('should fire callback when clicked', () => {
    const option = 'testOption';
    const callback = jest.fn();

    const wrapper = shallow(
      <DropdownItem option={option} onSelect={callback} />
    );
    wrapper.find('div.dropdown-item').simulate('click');

    expect(callback).toHaveBeenCalled();
    expect(callback).toHaveBeenCalledWith(option);
  });
});
