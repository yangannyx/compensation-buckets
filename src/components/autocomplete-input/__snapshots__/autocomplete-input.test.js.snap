// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteInput component should render 1`] = `
<div
  className="container"
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="dropdown"
  />
</div>
`;
